tools/flowc/backends/nim/natives/string/fromCharCode.nim

```diff
@@ -1,4 +1,7 @@
 import unicode
 
-func $F_0(fromCharCode)*(code: int32): string =
-  return unicode.toUTF8(cast[Rune](code))
\ No newline at end of file
+func $F_0(fromCharCode)*(code: int32): String =
+  when use16BitString:
+    return @[Utf16Char(cast[int16](code))]
+  else:
+    return unicode.toUTF8(cast[Rune](code))
```

Comment:
In Java the implementation uses the 32 bit code to make a uft16 string (so it can end up with a string of length 2)
In Cpp the implementation uses just cast to 16 bit, and use that as an single string of lenght 1.

This implementation does a mix of this. When use16BitString you only use the 16 bit, when not use16BitString you uses all the 32 bits. 

I suggest: 

when use16BitString:
    return rt_utf8_to_string(unicode.toUTF8(cast[Rune](code)));
  else:
    return unicode.toUTF8(cast[Rune](code))

Reply:
An relevant optimization would be when use16BitString and "code & 0xffff0000 = 0" then we can use the fast way: 
 @[Utf16Char(cast[int16](code))]

Reply:
Yes, on surrogate pairs this code will fail. I replaced with:
`    if (code < 0x0D800 or (0x0DFFF < code and code < 0xFFFF)):
      return @[Utf16Char(cast[int16](code))]
    else:
      # must be represented as a surrogate pair
      return rt_utf8_to_string(unicode.toUTF8(cast[Rune](code)))`